examples/seismic/model.py

```diff
@@ -191,36 +197,84 @@ class Model(GenericModel):
         Tilt angle in radian.
     phi : array_like or float
         Asymuth angle in radian.
-
-    The `Model` provides two symbolic data objects for the
-    creation of seismic wave propagation operators:
-
-    m : array_like or float
-        The square slowness of the wave.
-    damp : Function
-        The damping field for absorbing boundary condition.
+    b : array_like or float
+        Buoyancy
+    rho : array_like or float
+        Density
+    vs : array_like or float
+        S-wave velocity
+    qp : array_like or float
+        P-wave attenuation
+    qs : array_like or float
+        S-wave attenuation
+    lame: Bool
+        Whether to use Lame parameter (default) or vp/vs
     """
+    _known_parameters = ['vp', 'damp', 'vs', 'b', 'epsilon', 'delta',
+                         'theta', 'phi', 'qp', 'qs', 'lam', 'mu']
+
     def __init__(self, origin, spacing, shape, space_order, vp, nbl=20,
-                 dtype=np.float32, epsilon=None, delta=None, theta=None, phi=None,
-                 subdomains=(), damp_mask=False, **kwargs):
-        super(Model, self).__init__(origin, spacing, shape, space_order, nbl, dtype,
-                                    subdomains, damp_mask=damp_mask, **kwargs)
+                 dtype=np.float32, subdomains=(), bcs="mask", grid=None, **kwargs):
+        super(SeismicModel, self).__init__(origin, spacing, shape, space_order, nbl,
+                                           dtype, subdomains, grid=grid, bcs=bcs)
 
-        # Create square slowness of the wave as symbol `m`
-        self._vp = self._gen_phys_param(vp, 'vp', space_order)
+        # Initialize physics
+        self._initialize_physics(vp, space_order, **kwargs)
 
-        # Additional parameter fields for TTI operators
-        self.epsilon = self._gen_phys_param(epsilon, 'epsilon', space_order)
-        self.scale = 1 if epsilon is None else np.sqrt(1 + 2 * np.max(epsilon))
+        # User provided dt
+        self._dt = kwargs.get('dt')
 
-        self.delta = self._gen_phys_param(delta, 'delta', space_order)
-        self.theta = self._gen_phys_param(theta, 'theta', space_order)
-        if self.grid.dim > 2:
-            self.phi = self._gen_phys_param(phi, 'phi', space_order)
+    def _initialize_physics(self, vp, space_order, **kwargs):
+        """
+        Initialize physical parameters and type of physics from inputs.
+        The types of physics supportedare:
+        - acoustic: vp and b only
+        - elastic: vp + vs + b turn into lam/mu/b
+        - visco-acoustic: vp + b + qp
+        - visco-elastic: vp + vs + b + qs
+        - vti: epsilon + delta
+        - tti: epsilon + delta + theta + phi
+        """
+        params = []
+        # Buoyancy
+        b = kwargs.get('b', 1)
+
+        # Initialize elastic with Lame parametrization
+        if 'vs' in kwargs.keys():
+            vs = kwargs.get('vs')
```

Comment:
`vs` must be present or you'd get an exception in the line below. So just better to use bracket notation [] instead of .get